pkg/sr/serde.go

```diff
@@ -134,6 +131,20 @@ func (s *Serde) SetDefaults(opts ...SerdeOpt) {
 	s.defaults = opts
 }
 
+// SetHeader configures which header should be used when encoding and decoding
```

Comment:
wdyt about this this being `SetGlobal`, and then `SerdeHeader` being a `GlobalOpt` -- i.e. something that applies to all serdes? y/n?

Reply:
Sure, it would make it easy to add more options in the future. I think naming could be better though. IMO `SerdeOpt` sounds like a better fit for configuring `Serde` (i.e. global options), while the current options could be something like `RegisterOpt` (since they are used in `Register`). This naming would play well with the changes I proposed in the [other PR](https://github.com/twmb/franz-go/pull/466), which introduces `EncodeOpt`, we end up with:
- `SerdeOpt` configures `Serde`
- `RegisterOpt` configures `Serde.Register`
- `EncodeOpt` configures `Serde.Encode`

While we're at it, it might make sense to change `Opt` to `ClientOpt` and introduce `NewSerde(opts ...SerdeOpt)` to match `NewClient`. WDYT? I can remove `SetHeader` from this PR and introduce the new options in a separate PR, to not increase the scope too much.

Reply:
* +1 to adding NewSerde, which also removes the need for SetDefaults (so that can / should be removed).
* +1 to changing Opt to ClientOpt
* +1 to changing the current register opts to RegisterOpt, and then changing the new opt introduced here to SerdeOpt _however_ all RegisterOpts need to implement SerdeOpt so that they can be used as defaults.
* I'm not sure about the difference between RegisterOpt and EncodeOpt. I need to look at the other PR. I remember glancing at it and not being sure why it was split. Also I wonder if EncodingOpt is better than both RegisterOpt and EncodeOpt (although I could favor EncodeOpt for brevity)

Reply:
Also I'm sorry this has been such a slow review cycle, to put it short: I was involved with something that was time consuming sun up to sun down and also went on for ~3w (coincidentally, right when you opened the PR) -- and I'm trying to not be inside on the weekends.

Reply:
No worries! I'll get back to this PR in a day or two 👍

Reply:
I might have time on Thursday to take this over if you're not able to get to it (I admit that this review cycle has gone on for ... quite a while).

Reply:
So the situation turned around, because I'm pretty busy now and I'm not sure I'll get a chance to look into this before the weekend. If you get the time, feel free to take over the work that's left.